### PR TITLE
Update Synopsys references to blackduck

### PR DESCRIPTION
**Pull request by devmehtabd**
Update all the synopsys references to blackduck The PR includes all the following changes:

- Copyrights
- Gradle Group names
- Package Names
- Sonar cloud endpoint

Note: One more commit will be pushed to this PR after integration-common library is released to update import statement and check if the build is passing.

### CODE REVIEW

File: src/main/java/com/blackduck/integration/phonehome/PhoneHomeClient.java [hunk 1]
```diff
@@ -74,25 +71,15 @@ public void postPhoneHomeRequest(PhoneHomeRequestBody phoneHomeRequestBody, Map<
...
     private String checkOverridePhoneHomeUrl(Map<String, String> environmentVariables) {
-        String overrideUrl;
-
-        overrideUrl = environmentVariables.get(PHONE_HOME_URL_OVERRIDE_VARIABLE);
-        if (StringUtils.isBlank(overrideUrl)) {
-            overrideUrl = environmentVariables.get(BLACKDUCK_PHONE_HOME_URL_OVERRIDE_VARIABLE);
-        }
-
-        return overrideUrl;
+        return environmentVariables.get(BLACKDUCK_PHONE_HOME_URL_OVERRIDE_VARIABLE);
```
Comment (DanaMaxfield) [inline src/main/java/com/blackduck/integration/phonehome/PhoneHomeClient.java]: What is the reason for these changes?

Comment (devmehtabd) [reply src/main/java/com/blackduck/integration/phonehome/PhoneHomeClient.java]: The earlier properties had SYNOPSYS in their names so I figured we would be removing support for them as well, which is the reason for these code changes. If not I can revert them.

**Review by andrian-sevastyanov — APPROVED** [review]

File: src/main/java/com/blackduck/integration/phonehome/PhoneHomeClient.java (hunk 1 above)
Comment (niravrsynopsys) [reply src/main/java/com/blackduck/integration/phonehome/PhoneHomeClient.java]: This change feels a bit intrusive but since we do need to rename SYNOPSYS_SKIP_PHONE_HOME to a new name, here are my thoughts on how this can still work.

I think in the existing code, `SYNOPSYS_SKIP_PHONE_HOME` and `BLACKDUCK_SKIP_PHONE_HOME` were meant for 2 different purposes. 

There are 2 ways analytics can be disabled:
1) By setting the `SYNOPSYS_SKIP_PHONE_HOME=true` environment variable. See [here](https://sig-product-docs.synopsys.com/bundle/integrations-detect/page/troubleshooting/usage-metrics.html).
2) By setting `ANALYTICS=false` in blackduck-config.env in Hub. This setting takes priority over the former one. See [here](https://sig-product-docs.synopsys.com/bundle/bd-hub/page/Install_Swarm/conf_analytics.html).

My understanding is that `BLACKDUCK_SKIP_PHONE_HOME` was just an env variable that was meant to correspond with  the `ANALYTICS` env variable in BlackDuck.

But in case of Detect, after reading BlackDuck's analytics config, Detect doesn't set `BLACKDUCK_SKIP_PHONE_HOME`. First, Detect calls the `/api/internal/integration-settings/analytics` endpoint and retrieves the value of the `ANALYTICS` variable configured in Hub. It then simply skips initiating phone home if that value was false. It doesn't update any env variable for phone-home-client's decisions.

So the replacement here of the original OR condition with `environmentVariables.containsKey(BLACKDUCK_SKIP_PHONE_HOME_VARIABLE)` would be safe enough to do as long as:
1) Neither Alert nor Detect is setting/reading the `BLACKDUCK_SKIP_PHONE_HOME` variable,
AND
2) Detect documentation is updated such that `SYNOPSYS_SKIP_PHONE_HOME` property is deprecated/removed and a new property named `BLACKDUCK_SKIP_PHONE_HOME` is introduced.

Comment (devmehtabd) [reply src/main/java/com/blackduck/integration/phonehome/PhoneHomeClient.java]: I see, thank you for the brief explanation. I think I understand the solution you have provided. Let's confirm this with @DanaMaxfield tomorrow in the follow up meeting and then we can decide what we want to go ahead with.

**Review by andrian-sevastyanov — APPROVED** [review]

**Review by niravrsynopsys — APPROVED** [review]

**Review by DanaMaxfield — APPROVED** [review]
